lib/child_process.js

```diff
@@ -230,7 +230,11 @@ exports.execFile = function(file /* args, options, callback */) {
   }
 
   child.stdout.addListener('data', function(chunk) {
-    stdoutLen += chunk.length;
+    if (!encoding) {
```

Comment:
According to the docs, the unit of `maxBuffer` is bytes, so I think these should unconditionally use `Buffer.byteLength()`.